### PR TITLE
Crystal support

### PR DESCRIPTION
**Pull request by theseankelly**
This is a basic port to Crystal - intentionally avoided adding new features.

* API updates from bouncy to crystal
* Rebased all of @esteve's patches to their crystal equivalents
* Updated repo files and build instructions for crystal
* CMake scripts to pass over action IDL files

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,7 +98,7 @@ vcs import src < ros2_dotnet_uwp.repos
 cd \dev\ament
 call install\local_setup.bat
 cd \dev\ros2
-colcon build --merge-install --cmake-args -A "%TARGET_ARCH%" -DCMAKE_SYSTEM_NAME=WindowsStore -DCMAKE_SYSTEM_VERSION=10.0.14393 -DTHIRDPARTY=ON -DINSTALL_EXAMPLES=OFF -DBUILD_TESTING=OFF
+colcon build --merge-install --packages-ignore rcl_logging_log4cxx --cmake-args -A "%TARGET_ARCH%" -DCMAKE_SYSTEM_NAME=WindowsStore -DCMAKE_SYSTEM_VERSION=10.0.17763 -DTHIRDPARTY=ON -DINSTALL_EXAMPLES=OFF -DBUILD_TESTING=OFF
```
Comment (esteve) [inline README.md]: Would this version work with the Hololens? I used 14393 because at the time, the Hololens used a rather old version of the Windows SDK.

Comment (esteve) [reply README.md]: Awesome, thanks! Then that means that the C# side can also use newer features?

**Comment by esteve** [comment]
@theseankelly this is what I'm going to do: I'll create a ros2-dotnet organization, move the repos there and add you as a member. Then you won't have to keep the changes in your personal fork. Is that ok?

**Comment by theseankelly** [comment]
@esteve sounds good!

**Comment by esteve** [comment]
@theseankelly done! I just sent you the invitation to join the team, welcome aboard :-) Feel free to create the forks of the ROS2 repos (i.e. rcl, rmw, etc.) needed for building ROS2 for UWP.

**Comment by theseankelly** [comment]
Thanks, sir! I have to run but I'll do that (and clean up my repos files accordingly) hopefully this evening.

File: README.md (hunk 1 above)
Comment (esteve) [reply README.md]: I recall having to use some old C# API, but unfortunately I can't remember where the code was. I think it had to do with reflection or something like that.

**Comment by esteve** [comment]
@theseankelly no worries, thanks for all the hard work!

**Comment by theseankelly** [comment]
Alright @esteve, I've updated the repos. Does CI work for the organization? Doesn't look like it's kicked off. I haven't fully tested the build on all architectures yet.

**Comment by esteve** [comment]
@theseankelly the CI is still using my repo for finding the `azure-pipelines.yml` file, which no longer exists. I can't find a way to change the path of the pipelines YAML file, so I think I'll have to recreate the pipelines for this repository.

**Comment by esteve** [comment]
@theseankelly sorry for the slow response. Do you have a username on Azure pipelines? I don't really know how permissions work there, so I'm not sure how to add you anyway. Do you have any tips? Thanks!

**Comment by theseankelly** [comment]
@esteve No worries, I've been veeeerry slowly plunking away at dashing support (the new idl format mostly) in the meantime. Just created an account with my email. Unclear what my 'username' is though..

**Comment by theseankelly** [comment]
@esteve FYI I've made a couple more updates discovered while porting to Dashing. I've manually tested the build for Desktop x64 and UWP x86/x64/ARM.

I've also got Dashing support completed and ready for PR as soon as this goes through.

**Comment by theseankelly** [comment]
closed and reopened to trigger CI

**Comment by theseankelly** [comment]
FYI CI passed for all flavors

**Pull request merged**